src/platform/utilities/api/index.js

```diff
@@ -1,8 +1,62 @@
 import * as Sentry from '@sentry/browser';
 
 import environment from '../environment';
+import ENVIRONMENTS from '../../../site/constants/environments';
 import localStorage from '../storage/localStorage';
 
+const SSOeEndpoint = () => {
+  let environmentPrefix;
+  switch (ENVIRONMENTS.BUILDTYPE) {
+    case ENVIRONMENTS.VAGOVSTAGING:
+      environmentPrefix = 'sqa.';
+      break;
+    case ENVIRONMENTS.VAGOVDEV:
+      environmentPrefix = 'int.';
+      break;
+    default:
+      environmentPrefix = '';
+  }
+
+  return `https://${environmentPrefix}eauth.va.gov/keepalive`;
+};
+
+function refreshSSOeSession(sessionExpiration) {
+  const minutesUntilExpiration =
+    (sessionExpiration.getTime() - Date.now()) / (1000 * 60);
+
+  if (minutesUntilExpiration > 10) {
+    // We want to minimize external keepalive calls, so this functions
+    // as self-enforced rate limiting
+    return;
+  }
+
+  fetch(SSOeEndpoint(), {
+    method: 'GET',
+    credentials: 'include',
+    cache: 'no-store',
+  })
+    .then(res => {
+      const hasSSOsession = res.headers.get('session-alive');
+
+      if (hasSSOsession) {
+        const sessionTimeout = res.headers.get('session-timeout');
+        const expirationTime = new Date();
+
+        // sessionTimeout is in seconds, convert to milliseconds and add from current time
+        expirationTime.setTime(
+          expirationTime.getTime() + sessionTimeout * 1000,
+        );
+        localStorage.setItem('sessionExpirationSSO', expirationTime);
+      }
+    })
+    .catch(err => {
+      Sentry.withScope(scope => {
+        scope.setExtra('error', err);
+        Sentry.captureMessage(`SSOe error: ${err.message}`);
+      });
+    });
+}
+
```

Comment:
This adds more SSO specific logic than I anticipated here. I can move this to an adjacent `SSOutilities.js` if it makes more sense to keep this cleaner.